BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -0,0 +1,400 @@
+        guard let jsonVal = self.commonJsonValueGetter() else { return nil }
+        if self.itemType == "HKBiologicalSex" {
+            return (self.value as? HKBiologicalSex)?.demographicDataValue
+        }
+        
+        return jsonVal
+    }
+    
+    func commonCheckTypeCompatible(newValue: Any?) -> Bool {
+        guard newValue != nil else { return true }
+        
+        switch self.itemType {
+        case "String":
+            return newValue as? String != nil
+            
+        case "Number":
+            return newValue as? NSNumber != nil
+            
+        case "Bool":
+            return newValue as? NSNumber != nil
+            
+        case "Date":
+            return newValue as? NSDate  != nil
+            
+        case "HKBiologicalSex":
+            return newValue as? HKBiologicalSex != nil
+            
+        case "HKQuantity":
+            return newValue as? NSNumber != nil
+            
+        default:
+            return false
+        }
+    }
+}
+
+class SBAProfileItemBase: NSObject, SBAProfileItem {
+    /**
+     The value property is used to get and set the profile item's value in whatever internal data
+     storage is used by the implementing class.
+     */
+    open var value: Any?
+
+    fileprivate let sourceDict: [AnyHashable: Any]
+    
+    open var key: String {
+        get {
+            return sourceDict["key"]! as! String
+        }
+    }
+    
+    open var sourceKey: String {
+        get {
+            return sourceDict["sourceKey"] as! String? ?? self.key
+        }
+    }
+    
+    open var demographicKey: String {
+        get {
+            return sourceDict["demographicKey"] as! String? ?? self.key
+        }
+    }
+    
+    open var itemType: String {
+        get {
+            return sourceDict["itemType"]! as! String
+        }
+    }
+    
+    public required init(dictionaryRepresentation dictionary: [AnyHashable: Any]) {
+        sourceDict = dictionary
+        super.init()
+    }
+    
+    open var jsonValue: SBBJSONValue? {
+        get {
+            return self.commonJsonValueGetter()
+        }
+        
+        set {
+            commonJsonValueSetter(value: newValue)
+        }
+    }
+    
+    open var demographicJsonValue: SBBJSONValue? {
+        get {
+            return self.commonDemographicJsonValue()
+        }
+    }
+}
+
+class SBAKeychainProfileItem: SBAProfileItemBase {
+    private var keychain: SBAKeychainWrapper
+    
+    public required init(dictionaryRepresentation dictionary: [AnyHashable: Any]) {
+        keychain = SBAUser.shared.keychain
+        
+        let keychainService = dictionary["keychainService"] as! String?
+        let keychainAccessGroup = dictionary["keychainAccessGroup"] as! String?
+        
+        if (keychainService != nil || keychainAccessGroup != nil) {
+            keychain = SBAKeychainWrapper(service: keychainService, accessGroup: keychainAccessGroup)
+        }
```

Comment:
````
self.keychain = {
    let keychainService = dictionary["keychainService"] as? String
    let keychainAccessGroup = dictionary["keychainAccessGroup"] as? String
    guard (keychainService != nil || keychainAccessGroup != nil) 
    else {
        return SBAUser.shared.keychain
    }
    return SBAKeychainWrapper(service: keychainService, accessGroup: keychainAccessGroup)
}()
````

OR

````
    let keychainService = dictionary["keychainService"] as? String
    let keychainAccessGroup = dictionary["keychainAccessGroup"] as? String
    self.keychain =  (keychainService != nil || keychainAccessGroup != nil) ?
        SBAKeychainWrapper(service: keychainService, accessGroup: keychainAccessGroup) :
        SBAUser.shared.keychain
````

Either implementation will allow you to use `let` rather than `var` which is theoretically more performant and more importantly, will document that this is a property that is readonly and set on init.